src/metpy/plots/_util.py

```diff
@@ -283,3 +283,66 @@ def normalize(x):
     except TypeError:
         res = cols[normalize(c)]
     return res
+
+
+def plot_local_extrema(ax, extreme_vals, symbol, plot_val=True, **kwargs):
+    """Plot the local extreme (max/min) values of an array.
+
+    The behavior of the plotting will have the symbol horizontal/vertical alignment
+    be center/bottom and any value plotted will be center/top. The text size of plotted
+    values is 0.65 of the symbol size.
+
+    Parameters
+    ----------
+    ax : `matplotlib.axes`
+        The axes which to plot the local extrema
+    extreme_vals : `xarray.DataArray`
+        The DataArray that contains the variable local extrema
```

Comment:
Nothing other than I was creating this with declarative in mind, but I see how dropping this back to not using it would make it more widely applicable and not a pain to deal with in declarative.

Reply:
That's what i figured.